macOS/DuckDuckGo/TabBar/ViewModel/TabCollectionViewModel.swift

```diff
@@ -257,11 +257,15 @@ final class TabCollectionViewModel: NSObject {
     }
 
     @discardableResult func select(tab: Tab, forceChange: Bool = false) -> Bool {
-        guard let index = tabCollection.tabs.firstIndex(where: { $0 == tab }) else {
-            return false
+        if let index = tabCollection.tabs.firstIndex(where: { $0 == tab }) {
+            return selectUnpinnedTab(at: index, forceChange: forceChange)
+        }
+
+        if let index = pinnedTabsCollection?.tabs.firstIndex(where: { $0 == tab }) {
+            return selectPinnedTab(at: index, forceChange: forceChange)
         }
 
-        return selectUnpinnedTab(at: index, forceChange: forceChange)
+        return false
```

Comment:
I've noticed this API didn't have support for Tabs in the `pinnedTabsCollection`.

This change comes with batteries 🔋 and new Unit Tests!